js/customize-snapshots.js

```diff
@@ -1,104 +1,121 @@
-				snapshot.editControlSettings.create( 'title', snapshot.data.title );
-				snapshot.editControlSettings.create( 'date', snapshot.data.publishDate );
+				api.state.bind( 'change', function() {
+					if ( api.state( 'activated' ).get() && 'pending' === api.state( 'selectedChangesetStatus' ).get() ) {
+						if ( api.state( 'saved' ).get() && api.state( 'selectedChangesetStatus' ).get() === api.state( 'changesetStatus' ).get() ) {
+							saveBtn.val( snapshot.data.i18n.pendingSaved );
+						} else {
+							saveBtn.val( snapshot.data.i18n.savePending );
+						}
+					}
+				} );
 
-				api.bind( 'change', function() {
-					api.state( 'snapshot-submitted' ).set( false );
+				api.section( 'publish_settings', function( section ) {
+					snapshot.addPendingToStatusControl();
+					snapshot.addTitleControl( section );
+					snapshot.addInspectChangesetControl( section );
 				} );
 
+				// For backward compat.
 				snapshot.frontendPreviewUrl = new api.Value( api.previewer.previewUrl.get() );
 				snapshot.frontendPreviewUrl.link( api.previewer.previewUrl );
-				snapshot.isNotSavedPreviewingTheme = false;
-
-				snapshot.addButtons();
-				snapshot.editSnapshotUI();
-				snapshot.prefilterAjax();
 
 				api.trigger( 'snapshots-ready', snapshot );
 			} );
 
 			api.bind( 'save', function( request ) {
+				request.done( function( response ) {
+					if ( response.edit_link ) {
+						api.state( 'changesetInspectLink' ).set( response.edit_link );
+					}
+					if ( response.title ) {
+						api.state( 'changesetTitle' ).set( response.title );
+					}
 
-				request.fail( function( response ) {
-					var id = '#snapshot-dialog-error',
-						snapshotDialogPublishError = wp.template( 'snapshot-dialog-error' );
+					// Trigger an event for plugins to use, for backward compat.
+					api.trigger( 'customize-snapshots-update', response );
+				} );
+			} );
+		},
 
-					if ( response.responseText ) {
+		/**
+		 * Add title control to publish settings section.
+		 *
+		 * @param {wp.customize.Section} section Publish settings section.
+		 * @return {void}
+		 */
+		addTitleControl: function( section ) {
+		    var snapshot = this, titleControl, toggleControl;
 
-						// Insert the dialog error template.
-						if ( 0 === $( id ).length ) {
-							$( 'body' ).append( snapshotDialogPublishError( {
-								title: snapshot.data.i18n.publish,
-								message: api.state( 'snapshot-exists' ).get() ? snapshot.data.i18n.permsMsg.update : snapshot.data.i18n.permsMsg.save
-							} ) );
-						}
+			titleControl = new api.Control( 'changeset_title', {
+				type: 'text',
+				label: snapshot.data.i18n.title,
+				section: section.id,
+				setting: api.state( 'changesetTitle' ),
+				priority: 31
+			} );
 
-						snapshot.spinner.removeClass( 'is-active' );
+			api.control.add( titleControl );
 
-						$( id ).dialog( {
-							autoOpen: true,
-							modal: true
-						} );
-					}
-				} );
+			toggleControl = function( status ) {
```

Comment:
This can be removed. The title field should always be shown. I should be able to open the Customizer, make a change, provide a title (commit message), and then publish.